### PR TITLE
[[ Bugfix 21066 ]] Modify libUrlSetDriver to ensure behavior is not set while behavior is running

### PR DESCRIPTION
**Pull request by techstrategies**

### CODE REVIEW

File: ide-support/revliburl.livecodescript [hunk 1]
```diff
@@ -89,46 +89,52 @@ on libUrlSetDriver pDriver
...
-      unloadExternalDriver
+      unloadExternalDriver false
+      put pDriver into lvExtDriver
 
       if pDriver is not empty then
-         return loadExternalDriver(pDriver)
+         send loadExternalDriver to me in 0
       else
-         return empty
+         send libUrlSetBehavior to me in 0
```
Comment (montegoulding) [inline ide-support/revliburl.livecodescript]: @techstrategies these commands should be quoted for explicit vars

**Comment by montegoulding** [comment]
@techstrategies I'm wondering what we should do about a series of commands like this:
```
dispatch "revUnloadLibrary" to "tsNetLibURL"
libURLSetDriver <my driver>
get url <url>
```

**Comment by techstrategies** [comment]
@montegoulding Thanks for the review.  Looking back over it, the issue is only due to what occurs when unloading the driver (as that is when it is the behavior of revliburl).  When loading a new driver, the behavior can be set immediately without the "send".

My latest changes should work in a scenario with the commands you listed.  The dispatch "revUnloadLibrary" would stop libUrl using tsNet immediately by setting lvExtDriver to empty (even though it would still be attached as a behavior) and a "send" is issued to remove the behavior.

The next libUrlSetDriver would immediately set the behavior to the new driver successfully since that code is not being run from inside the existing behavior script (tsNetLibUrl) and it would also set lvExtDriver to the new driver accordingly.  After that the "get URL" command would use the new driver.

When libUrlSetBehavior finally gets called (from the "send" during the revUnloadLibrary to tsNet), lvExtDriver and the behavior of revLibUrl will both already be the newly loaded driver - so that "send" will do nothing.

File: ide-support/revliburl.livecodescript [hunk 2]
```diff
@@ -86,49 +86,58 @@ end libraryStack
...
    if lvExtDriver is not pDriver then
-      unloadExternalDriver
+      libUrlResetAll
+
+      unloadExternalDriver false
+      put pDriver into lvExtDriver
 
       if pDriver is not empty then
-         return loadExternalDriver(pDriver)
+         loadExternalDriver
```
Comment (montegoulding) [inline ide-support/revliburl.livecodescript]: Should we return the result here?

File: ide-support/revliburl.livecodescript [hunk 3]
```diff
@@ -86,49 +86,58 @@ end libraryStack
...
       else
-         return empty
+         send "libUrlSetBehavior" to me in 0
```
Comment (montegoulding) [inline ide-support/revliburl.livecodescript]: Maybe return empty after this

File: ide-support/revliburl.livecodescript [hunk 4]
```diff
@@ -86,49 +86,58 @@ end libraryStack
...
+command libUrlSetBehavior
+   if the behavior of me is not lvExtDriver then
+      set the behavior of me to lvExtDriver
+   end if
+end libUrlSetBehavior
+
 -- CW-2016-06-11: [[ External driver support ]] Call driver specific init/remove commands if external driver is in use.
 ## Send initialise command to external library
-private function loadExternalDriver pDriver
+command loadExternalDriver
```
Comment (montegoulding) [inline ide-support/revliburl.livecodescript]: I think this can go back to being private now if I'm not mistaken

**Comment by techstrategies** [comment]
@montegoulding Ah, yes.... sorry, just needed to fix that back up since the reason behind those changes were no longer needed.

**Comment by montegoulding** [comment]
@livecode-vulcan review ok ab64d07

**Comment by livecode-vulcan** [comment]
💙 review by @montegoulding ok ab64d07ff7b2ca26a901069a1d3dae95f8965673

**Comment by livecode-vulcan** [comment]
😎 test success ab64d07ff7b2ca26a901069a1d3dae95f8965673

* try-community-armv6-android-api8: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api8/builds/1342)
* try-community-armv6-android-api9: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api9/builds/1358)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/1332)
* try-community-universal-ios-iphoneos11.2: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos11.2/builds/285)
* try-community-universal-ios-iphonesimulator11.2: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator11.2/builds/283)
* try-community-universal-mac-macosx10.6: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.6/builds/1336)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/1362)
* try-community-x86-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian7/builds/1342)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/1356)
* try-community-x86_64-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian7/builds/1341)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/1351)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/1392)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/1332)

**Pull request merged**